### PR TITLE
develop a agent to make routes by watching eips

### PR DESCRIPTION
**Pull request by magicsong**
#4

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -95,3 +95,8 @@ metadata:
   name: webhook-server-secret
   namespace: system
 
+# ---
```
[the remark this answers is not in the document]
Comment (magicsong) [reply config/manager/manager.yaml]: these comments are temporary removed and will be used in next version

File: pkg/controller/service/service_controller.go [hunk 2]
```diff
@@ -120,116 +88,41 @@ func (r *ReconcileService) Reconcile(request reconcile.Request) (reconcile.Resul
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
+	origin := instance.DeepCopy()
+	reconcileLog := log.WithValues("Service Name", instance.GetName(), "Namespace", instance.GetNamespace())
 	needReconile, err := r.useFinalizerIfNeeded(instance)
```
Comment (wnxn) [inline pkg/controller/service/service_controller.go]: Reconcile

File: pkg/strategy/strategy.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+func (defaultStrategy) Select(serv *corev1.Service, eips *v1alpha1.EIPList) (*v1alpha1.EIP, error) {
+	if len(eips.Items) == 0 {
+		return nil, errors.NewResourceNotEnoughError("eip")
+	}
+	for _, ip := range eips.Items {
+		if !ip.Spec.Disable && !ip.Status.Occupied {
+			return &ip, nil
+		}
+	}
+	return nil, fmt.Errorf("No suitable ip has empty ports for service %s", serv.Name)
```
Comment (wnxn) [inline pkg/strategy/strategy.go]: No enough EIP resource for allocation to Service: [%s], serv.Name

File: pkg/strategy/strategy.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+type portBasedStrategy struct {
+}
+
+func (portBasedStrategy) Select(serv *corev1.Service, eips *v1alpha1.EIPList) (*v1alpha1.EIP, error) {
```
Comment (wnxn) [inline pkg/strategy/strategy.go]: Should add unit test to test EIP allocation algorithm

Comment (magicsong) [reply pkg/strategy/strategy.go]: #7 will do that

**Comment by magicsong** [comment]
Some problems have moved to  issues, ref #12, #10, #7
Let this pr approved now.

**Pull request merged**